File: app/src/main/java/com/vincent/githubusersearching/model/ItemSearchResult.java
package com.vincent.githubusersearching.model;

import com.vincent.githubusersearching.AppController;

import java.util.List;

public class ItemSearchResult {

    private int total_count;
    private List<ItemUser> items;

    public int getTotalCount() {
        return total_count;
    }

    public List<ItemUser> getUserList() {
        return items;
    }

    public static class ItemUser {

        private String login;
        private String avatar_url;
        private String html_url;

        private int position;

        public String getName() {
            return login;
        }

        public String getAvatarUrl() {
            return avatar_url;
        }

        public void openWebPage() {
            AppController.getInstance().openWebWithBrowser(html_url);
        }

        public int getPosition() {
            return position;
        }

        public void setPosition(int position) {
            this.position = position + 1;
        }
    }
}

File: app/src/main/java/com/vincent/githubusersearch/model/ItemSearchResult.kt
package com.vincent.githubusersearch.model

import com.google.gson.annotations.SerializedName
import com.vincent.githubusersearch.AppController

class ItemSearchResult(@SerializedName("total_count") var totalCount: Int, @SerializedName("items") var userList: List<GitHubUser>) {

    inner class GitHubUser(
            private var id: Long,
            private var login: String,
            private var avatar_url: String,
            private var html_url: String) : User {

        private var position: Int = 0

        override fun getId(): Long {
            return id
        }

        override fun getName(): String {
            return login
        }

        override fun getAvatarUrl(): String {
            return avatar_url
        }

        override fun getWebUrl(): String {
            return html_url
        }

        override var indexNumber: Int
            get() = position
            set(value) {
                this.position = value + 1
            }

        override fun onPerformAction() {
            AppController.instance.openWebWithBrowser(html_url)
        }
    }
}
